docs/site/migration/models/mixins.md

```diff
@@ -6,8 +6,805 @@ sidebar: lb4_sidebar
+import {Constructor} from '@loopback/context';
+
+import {Model} from '@loopback/repository';
+import {FindByTitleInterface} from './findByTitleInterface';
+
+export function FindByTitleRepositoryMixin<
+  M extends Model,
+  T extends Constructor<any>
+>(superClass: T) {
+  return class extends superClass implements FindByTitleInterface<M> {
+    async findByTitle(title: string): Promise<M[]> {
+      const titleFilter = {
+        where: {
+          title: title,
+        },
+      };
+      const foundItems = await this.find(titleFilter);
+      return foundItems;
+    }
+  };
+}
+```
+
+### Generating A Repository Via The CLI
+
+A CLI-generated repository for a model `Note` would look like this:
+
+{% include code-caption.html content="src/repositories/note.repository.ts" %}
+
+```ts
+export class NoteRepository extends DefaultCrudRepository<
+  Note,
+  typeof Note.prototype.id,
+  NoteRelations
+> {
+  constructor(@inject('datasources.db') dataSource: DbDataSource) {
+    super(Note, dataSource);
+  }
+}
+```
+
+### Adjust NoteRepository To Use FindByTitleRepositoryMixin
+
+Import the `FindByTitleRepositoryMixin` mixin class factory function, and adjust
+the declaration of the `NoteRepository` class to extend the class returned from
+the mixin function that takes in the `DefaultCrudRepository` superclass an
+input.
+
+{% include code-caption.html content="src/repositories/note.repository.ts" %}
+
+```ts
+import {FindByTitleRepositoryMixin} from '../mixins/findByTitleRepositoryMixin';
+import {DefaultCrudRepository} from '@loopback/repository';
+import {Note, NoteRelations} from '../models';
+import {DbDataSource} from '../datasources';
+import {inject, Constructor} from '@loopback/core';
+
+export class NoteRepository extends FindByTitleRepositoryMixin<
+  Note,
+  Constructor<
+    DefaultCrudRepository<Note, typeof Note.prototype.id, NoteRelations>
+  >
+>(DefaultCrudRepository) {
+  constructor(@inject('datasources.db') dataSource: DbDataSource) {
+    super(Note, dataSource);
+  }
+}
+```
+
+We have now added the `findByTitle` method to a repository via a mixin class
+factory function.
+
+### Defining A Controller Mixin Class Factory Function
+
+In **src/mixins/findByTitleControllerMixin.ts**, let's define the mixin class
+factory function `FindByTitleControllerMixin` which adds the `findByTitle`
+method to any controller.
+
+{% include code-caption.html content="src/mixins/src/mixins/findByTitleControllerMixin.ts" %}
+
+```ts
+import {Constructor} from '@loopback/context';
+import {Model} from '@loopback/repository';
+import {FindByTitleInterface} from './findByTitleInterface';
+import {param, get, getModelSchemaRef} from '@loopback/rest';
+
+export interface FindByTitleControllerMixinParms {
+  basePath: string;
+  modelClass: Constructor<{}>;
+  modelClassName: string;
+}
+
+export function FindByTitleControllerMixin<
+  M extends Model,
+  T extends Constructor<any>
+>(superClass: T, parms: FindByTitleControllerMixinParms) {
```

Comment:
I believe we usually abbreviate `parameters` as `params` (i.e. `params` argument, `FindByTitleControllerMixinParams` interface).

Having said that, we tend to use the name `options` instead of `params`.

```suggestion
>(superClass: T, options: FindByTitleControllerMixinOptions) {
```